internal/lib/stats/stats_server.go

```diff
@@ -130,6 +147,174 @@ func (ss *StatsServer) updateSandbox(sb *sandbox.Sandbox) *types.PodSandboxStats
 	return sandboxStats
 }
 
+// GenerateSandboxContainerMetrics generates a list of metrics for the specified sandbox
+// containers by collecting metrics from different sources based on the includedMetrics
+// except for network metrics which is collected at pod level.
+func (ss *StatsServer) GenerateSandboxContainerMetrics(sb *sandbox.Sandbox, c *oci.Container, sm *SandboxMetrics) []*types.Metric {
+	var metrics []*types.Metric
+
+	var sbCgroupStats *cgroups.Stats
+	var err error
+
+	includedMetrics := ss.Config().IncludedPodMetrics
+	var updatedIncludedMetrics []string
+	// Skip network metrics as they are collected at pod level
+	for _, val := range includedMetrics {
+		if val != "network" {
+			updatedIncludedMetrics = append(updatedIncludedMetrics, val)
+		}
+	}
+	// Create a cgroup manager for the container if it doesn't exist
+	if ss.ctrCgMgrs[c.ID()] == nil {
+		if err := ss.updateCgMgrs(sb, c); err != nil {
+			logrus.Errorf("Unable to Create a cgroup manager for the container %s: %v", c.ID(), err)
+			return nil
+		}
+	}
```

Comment:
since we're on the same container here, we can initialize the cgmgr here and then use it throughout, instead of duplicating the map lookup